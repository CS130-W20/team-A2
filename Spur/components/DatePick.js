import React, {useState} from 'react';
import {View, Button, Platform, StyleSheet} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';

const DatePick = (props) => {
  const [date, setDate] = useState(new Date(1598051730000));
  const [mode, setMode] = useState(props.type);
  const [show, setShow] = useState(false);

  const showMode = currentMode => {
    setShow(true);
    setMode(currentMode);
  };

  const showDatepicker = () => {
    showMode(props.type);
    setShow(!show);
    props.onChange(date);
  };

  const showTimepicker = () => {
    showMode('time');
  };

  const onChange = (event, selectedDate) => {

    console.log(selectedDate);

    props.onChange(selectedDate);
    const currentDate = selectedDate || date;
    
    setDate(currentDate);
    setShow(Platform.OS === 'ios' ? true : false);

    
  };

  return (
    <View>
      <View>
        <Button onPress={showDatepicker} title={props.text} />
      </View>
      {show && (
        <DateTimePicker
          testID="dateTimePicker"
          timeZoneOffsetInMinutes={-7 * 60}
          value={date}
          mode={mode}
          is24Hour={false}
          display="default"
          onChange={onChange}
        />
      )}
    </View>
  );
};

export default DatePick;