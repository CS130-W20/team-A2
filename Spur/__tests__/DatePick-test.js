import * as React from 'react';
import renderer from 'react-test-renderer';
import DatePick from '../components/DatePick';

describe(`Date Pick Tester`, () => {
    test('Date Displays correctly', () => {
    
        const time = {
            month: 3,
            day: 20,
            hours: 16,
            minutes: 30
        }

        const date = renderer.create(<DatePick type='date' time={time} />).toJSON();
        expect(date).toMatchSnapshot();



    });

    test('Time Displays correctly', () => {
    
        const time = {
            month: 3,
            day: 20,
            hours: 16,
            minutes: 30
        }

        const timepick = renderer.create(<DatePick type='time' time={time} />).toJSON();
        expect(timepick).toMatchSnapshot();


    });

  });