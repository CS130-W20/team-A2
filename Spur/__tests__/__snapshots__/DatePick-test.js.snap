// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Pick Tester Date Displays correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onFocus={[Function]}
          placeholder="3/20"
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Date Pick Tester Time Displays correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onFocus={[Function]}
          placeholder="4:30 pm"
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
